core/frontend/src/tools/MeasureTool.ts

```diff
@@ -24,11 +24,24 @@ import { PrimitiveTool } from "./PrimitiveTool";
 import { BeButtonEvent, BeModifierKeys, CoreTools, EventHandled, InputSource } from "./Tool";
 import { ToolAssistance, ToolAssistanceImage, ToolAssistanceInputMethod, ToolAssistanceInstruction, ToolAssistanceSection } from "./ToolAssistance";
 import { GraphicType } from "../common/render/GraphicType";
+import { Format, FormatterSpec } from "@itwin/core-quantity";
 
 function translateBold(key: string) {
   return `<b>${CoreTools.translate(`Measure.Labels.${key}`)}:</b> `;
 }
 
+async function getFormatterSpecByKoQAndPersistenceUnit(koq: string, persistenceUnitName: string, fallbackQuantity: QuantityType): Promise<FormatterSpec | undefined> {
+  if (!IModelApp.formatsProvider) {
+    return IModelApp.quantityFormatter.findFormatterSpecByQuantityType(fallbackQuantity);
+  }
+  const formatProps = await IModelApp.formatsProvider.getFormat(koq);
+  if (undefined === formatProps)
+    return undefined;
+  const persistenceUnitProps = await IModelApp.quantityFormatter.unitsProvider.findUnitByName(persistenceUnitName);
+  const format = await Format.createFromJSON(koq, IModelApp.quantityFormatter.unitsProvider, formatProps)
+  return FormatterSpec.create(`${koq}_format_spec`, format, IModelApp.quantityFormatter.unitsProvider, persistenceUnitProps);
```

Comment:
It looks like this would need to be done by every possible consumer who wants to format value from persistence units to some format. Maybe quantity formatter could have a function that does this? E.g.:

```ts
async createFormatterSpec(
  props: {
    persistenceUnitName: string;
    formatProps: FormatProps;
    formatName?: string; // is this needed for a temporary format? maybe just assign some arbitrary string or Guid.createValue()?
  },
): Promise<FormatterSpec>;
```

Reply:
I like it. I've added a `createFormatterSpec` and `createParserSpec` helper methods inside `QuantityFormatter` that reflects the suggestion you made. formatName is optional in the interface I made for the args. This way we can also control the unitsProvider used will be the one inside `QuantityFormatter`.